DEHPCommon/UserPreferenceHandler/UserPreferenceService/UserPreferenceService.cs

```diff
@@ -0,0 +1,164 @@
+    public class UserPreferenceService<T> : IUserPreferenceService<T> where T : IUserPreference
+    {
+        /// <summary>
+        /// The logger for the current class
+        /// </summary>
+        private static Logger logger = LogManager.GetCurrentClassLogger();
+
+        /// <summary>
+        /// The path of the user preference directory storage
+        /// </summary>
+        public static string ApplicationExecutePath = Path.GetDirectoryName(Assembly.GetExecutingAssembly()?.Location);
+
+        /// <summary>
+        /// The name of folder where user preference is storage.
+        /// </summary>
+        public static string UserPreferenceDirectoryName = "UserPreferenceService";
+
+        /// <summary>
+        /// The setting file extension
+        /// </summary>
+        public const string SETTING_FILE_EXTENSION = ".settings.json";
+
+        /// <summary>
+        /// Gets or sets  user preference settings,
+        /// </summary>
+        public T UserPreferenceSettings { get; set; }
+
+        /// <summary>
+        /// Configuration user preference file path
+        /// </summary>
+        public string UserPreferenceDirectoryPath
+        {
+            get { return Path.Combine(ApplicationExecutePath, UserPreferenceDirectoryName); }
+        }
+
+        ///// <summary>
+        ///// Initializes a new instance of <see cref="UserPreferenceService"/>
+        ///// </summary>
+        //public UserPreferenceService()
+        //{
+        //    this.UserPreferenceSettings = new UserPreference();
+        //}
+
+        /// <summary>
+        /// Reads the <see cref="T"/> user preference in settings
+        /// </summary>
+        public void Read()
+        {
+            var assemblyName = this.QueryAssemblyTitle(typeof(T));
+
+            this.CheckConfigurationDirectory();
+
+            var path = Path.Combine(this.UserPreferenceDirectoryPath, assemblyName);
+
+            logger.Debug("Read user preference for {0} from {1}", assemblyName, path);
+
+            try
+            {
+                var fileExist = File.Exists($"{path}{SETTING_FILE_EXTENSION}");
+
+                if (!fileExist)
+                {
+                    this.Write((T)(new UserPreference() as IUserPreference));
+                }
+
+                var file = File.ReadAllText($"{path}{SETTING_FILE_EXTENSION}");
+                this.UserPreferenceSettings = JsonConvert.DeserializeObject<T>(file);
+            }
+            catch (Exception ex)
+            {
+                logger.Error(ex, "The user preference could not be read");
+
+                throw new IOException("The user preference could not be read", ex);
+            }
+        }
+
+        /// <summary>
+        /// Writes the <see cref="UserPreference"/> to disk
+        /// </summary>
+        /// <param name="userPreference">
+        /// The <see cref="UserPreference"/> that will be persisted
+        /// </param>
+        public void Write(T userPreference)
```

Comment:
Change name to `Save()` and remove the parameter. Just save `this.UserPreferenceSettings`